library/aoscx_upload_firmware.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (C) Copyright 2019-2020 Hewlett Packard Enterprise Development LP.
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'certified'
+}
+
+DOCUMENTATION = '''
+---
+module: aoscx_upload_firmware
+version_added: "2.8"
+short_description: Uploads a firmware image onto the AOS-CX switch.
+description:
+  - This module uploads a firmware image onto the switch stored locally or it can also upload
+    the firmware from an HTTP server.
+author: Aruba Networks (@ArubaNetworks)
+options:
+  partition_name:
+    description: Name of the partition for the image to be uploaded.
+    type: str
+    default: 'primary'
+    choices: ['primary', 'secondary']
+    required: false
+  firmware_file_path:
+    description: File name and path for locally uploading firmware image
+    type: str
+    required: false
+  remote_firmware_file_path:
+    description: "HTTP server address and path for uploading firmware image,
+      must be reachable through provided vrf
+      ex) http://192.168.1.2:8000/TL_10_04_0030A.swi"
+    type: str
+    required: false
+  config_type:
```

Comment:
there is no this option for module firmware ?

Reply:
when you're providing the `remote_firmware_file_path` variable the firmware is in that path:

```
- name: Upload firmware to primary through HTTP
  aoscx_upload_firmware:
    partition_name: 'primary'
    remote_firmware_file_path: 'http://192.168.1.2:8000/TL_10_04_0030P.swi'
    vrf: 'mgmt'
```